src/AlertBar.js

```diff
@@ -67,13 +69,15 @@ class AlertBar extends Component {
         }
     }
 
-    hide = () => {
+    hide = event => {
         clearTimeout(this.displayTimeout)
         this.setState({ visible: false })
 
         this.onHiddenTimeout = setTimeout(() => {
-            this.setState({ hidden: true })
-            this.props.onHidden && this.props.onHidden()
+            this.setState(
+                { hidden: true },
+                () => this.props.onHidden && this.props.onHidden({}, event)
+            )
```

Comment:
This is intentional: https://reactjs.org/docs/react-component.html#setstate

> Think of setState() as a request rather than an immediate command to update the component. For better perceived performance, React may delay it, and then update several components in a single pass. React does not guarantee that the state changes are applied immediately.

The second arg of `setState` is a function, which is called, after the state has been updated (which react does asynchronously).
If we put the call the way you proposed, the `onHidden` handler would be called before the component renders again.
In my opinion that's wrong. We don't know what apps are going to do in case they're using `onHidden` (e. g. counting the remaining dom elements? stupid example, but it demonstrates what I mean). I think the `onHidden` handler should be called after the component is actually hidden

Reply:
Oh yeah, neat! I've actually never worked that way before, but it makes complete sense. Since this is already a class based component, I'd prefer to have an internal onHidden handler and call that. To me that would be slightly more readable (and more efficient, but I don't think that's really an issue here).
I'll do a PR